### PR TITLE
Add all properties while creating a subscription

### PR DESCRIPTION
**Pull request by tcitworld**
Should fix issue #24469

This is my first PR, please don't eat me. :-)

### CODE REVIEW

**Comment by tcitworld** [comment]
And for now, MIT licensed.

**Comment by DeepDiver1975** [comment]
> This is my first PR, please don't eat me. :-)

No worries - THX

May I ask you to add unit tests? Ping me or @georgehrke on irc in case you need a helping hand.

> And for now, MIT licensed.

THX

**Comment by tcitworld** [comment]
Tests added.

**Comment by DeepDiver1975** [comment]
> Tests added.

THX @tcitworld - let's see what Jenkins says about it .... #25280

**Comment by DeepDiver1975** [comment]
@tcitworld please consider signing the CLA - this will grant you full access to the git repo and you can work on branches within the repo. CI will work then much easier - for all of us :wink:

**Comment by tcitworld** [comment]
Done.

**Comment by DeepDiver1975** [comment]
Jenkins is green on #25280

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@georgehrke mind testing? THX

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -1083,18 +1083,25 @@ function createSubscription($principalUri, $uri, array $properties) {
 			if (isset($properties[$xmlName])) {
 
 				$values[$dbName] = $properties[$xmlName];
-				$fieldNames[] = $dbName;
+			} else {
+				if ($dbName == 'calendarorder') {
```
Comment (georgehrke) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: Can you elaborate why this is necessary? it's not in https://github.com/tcitworld/core/blob/9fc27d1d649fa25ba7b4d88a021a7a1b71532c93/apps/dav/lib/CalDAV/CalDavBackend.php#LL1045-1049

Comment (georgehrke) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: sorry, i mean https://github.com/tcitworld/core/blob/9fc27d1d649fa25ba7b4d88a021a7a1b71532c93/apps/dav/lib/CalDAV/CalDavBackend.php#LL376-380

**Comment by georgehrke** [comment]
code looks good otherwise to me. Will test asap

File: apps/dav/lib/CalDAV/CalDavBackend.php (hunk 1 above)
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: I enforce default values to null (and 0 when database needs it). This part can be removed though.

**Comment by georgehrke** [comment]
```
<A:mkcol xmlns:A="DAV:">
    <A:set>
        <A:prop>
            <B:subscribed-strip-attachments xmlns:B="http://calendarserver.org/ns/" />
            <B:subscribed-strip-todos xmlns:B="http://calendarserver.org/ns/" />
            <A:resourcetype>
                <A:collection />
                <B:subscribed xmlns:B="http://calendarserver.org/ns/" />
            </A:resourcetype>
            <E:calendar-color xmlns:E="http://apple.com/ns/ical/">#1C4587FF</E:calendar-color>
            <A:displayname>Jewish holidays</A:displayname>
            <C:calendar-description xmlns:C="urn:ietf:params:xml:ns:caldav">Foo</C:calendar-description>
            <E:calendar-order xmlns:E="http://apple.com/ns/ical/">19</E:calendar-order>
            <B:source xmlns:B="http://calendarserver.org/ns/">
                <A:href>webcal://www.webcal.fi/cal.php?id=49&amp;format=ics&amp;wrn=1&amp;wp=1&amp;wf=26&amp;color=%231C4587&amp;cntr=us&amp;lang=en&amp;rid=wc</A:href>
            </B:source>
            <E:refreshrate xmlns:E="http://apple.com/ns/ical/">P1W</E:refreshrate>
            <B:subscribed-strip-alarms xmlns:B="http://calendarserver.org/ns/" />
        </A:prop>
    </A:set>
</A:mkcol>
```

- [ ] the database fields striptodos, stripalarms and stripattachments remain null.

**Comment by georgehrke** [comment]
@tcitworld I used https://chrome.google.com/webstore/detail/advanced-rest-client/hgmloofddffdnphfgcellkdfbfbjeloo?hl=en-US to send a `MKCOL` request using the mentioned payload and checked the db afterwords. Nothing fancy with Calendar.app.

**Comment by tcitworld** [comment]
The `$properties` array shows up like this for your request : 

```
Array
(
    [{http://calendarserver.org/ns/}subscribed-strip-attachments] => 
    [{http://calendarserver.org/ns/}subscribed-strip-todos] => 
    [{http://apple.com/ns/ical/}calendar-color] => #1C4587FF
    [{DAV:}displayname] => Jewish holidays
    [{urn:ietf:params:xml:ns:caldav}calendar-description] => Foo
    [{http://apple.com/ns/ical/}calendar-order] => 19
    [{http://calendarserver.org/ns/}source] => Sabre\DAV\Xml\Property\Href Object
        (
            [hrefs:protected] => Array
                (
                    [0] => webcal://www.webcal.fi/cal.php?id=49&format=ics&wrn=1&wp=1&wf=26&color=%231C4587&cntr=us&lang=en&rid=wc
                )

            [autoPrefix:protected] => 
        )

    [{http://apple.com/ns/ical/}refreshrate] => P1W
    [{http://calendarserver.org/ns/}subscribed-strip-alarms] => 
)
```

You'll notice that the `subscribed-strip-attachments` and `subscribed-strip-todos` keys exist even if they're not send through the request. It could be a SabreDAV issue, right ?

**Comment by DeepDiver1975** [comment]
> It could be a SabreDAV issue, right ?

I don't think so. These elements are part of the xml which is sent with the mkcol.
Sabre deserialized the xml to the properties array you are receiving

**Comment by georgehrke** [comment]
The problem here is `isset`. You need to check with array_key_exists for those strip properties

**Comment by tcitworld** [comment]
> The problem here is isset. You need to check with array_key_exists for those strip properties

That's what I was doing and I figured the keys always exist (see above). Sorry if I'm missing something.

**Comment by tcitworld** [comment]
Fixed.

**Comment by DeepDiver1975** [comment]
Since you are a contributor and have push access: please creat a branch in here and recreate the pr.thx

**Pull request closed without merging**